lib/wibox/layout/fixed.lua

```diff
@@ -333,61 +333,69 @@ end
 -- @param orig_width The available width.
 -- @param orig_height The available height.
 function fixed:fit(context, orig_width, orig_height)
-    local width_left, height_left = orig_width, orig_height
-    local spacing = self._private.spacing or 0
-    local widgets_nr = #self._private.widgets
-    local is_y = self._private.dir == "y"
-    local used_max = 0
-
-    -- when no widgets exist the function can be called with orig_width or
-    -- orig_height equal to nil. Exit early in this case.
-    if widgets_nr == 0 then
+    local widgets = self._private.widgets
+    local num_widgets = #widgets
+    -- Return early if there are no widgets to draw
+    if num_widgets < 1 then
         return 0, 0
     end
 
-    for k, v in pairs(self._private.widgets) do
-        local w, h = base.fit_widget(self, context, v, width_left, height_left)
-        local max
-
-        if is_y then
-            max = w
-            height_left = height_left - h
-        else
-            max = h
-            width_left = width_left - w
-        end
+    local width_left, height_left = orig_width, orig_height
+    local used_in_dir, used_max = 0, 0
+    local is_y = self._private.dir == "y"
+    local spacing = self._private.spacing or 0
 
-        if max > used_max then
-            used_max = max
-        end
+    for k, v in ipairs(widgets) do
+        -- Keep in mind that `spacing` may be negative to create overlap
+        if k > 1 and spacing ~= 0 then
+            used_in_dir = used_in_dir + spacing
 
-        if k < widgets_nr then
             if is_y then
                 height_left = height_left - spacing
             else
                 width_left = width_left - spacing
             end
         end
 
-        if width_left <= 0 or height_left <= 0 then
-            -- this complicated two lines determine whether we're out-of-space
-            -- because of spacing, or if the last widget doesn't fit in
-            if is_y then
-                 height_left = k < widgets_nr and height_left + spacing or height_left
-                 height_left = height_left < 0 and 0 or height_left
-            else
-                 width_left = k < widgets_nr and width_left + spacing or width_left
-                 width_left = width_left < 0 and 0 or width_left
+        local w, h = base.fit_widget(self, context, v, width_left, height_left)
+
+
+        -- Determine if the widget still fits
+        local is_enough
+
+        if is_y then
+            is_enough = h > 0 and height_left >= h
```

Comment:
```suggestion
            is_enough = height_left >= h
```
There may be a good reason for the `h > 0` condition needing to be true that I can't think of right now, but this does fix the issue for a widget with `visible = false`.